### PR TITLE
procyon-decompiler 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by middagj**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

As promised in #79869, the formula was removed in the meanwhile.

### CODE REVIEW

File: Formula/procyon-decompiler.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  depends_on "openjdk"
+
+  def install
+    libexec.install "procyon-decompiler-#{version}.jar"
+    (bin/"procyon-decompiler").write <<~EOS
+      #!/bin/bash
+      export JAVA_HOME="${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"
+      exec "${JAVA_HOME}/bin/java" -jar "#{libexec}/procyon-decompiler-#{version}.jar" "$@"
+    EOS
```
Comment (carlocab) [inline Formula/procyon-decompiler.rb:16]: Doesn't `write_jar_script` work?

Comment (middagj) [reply Formula/procyon-decompiler.rb]: Yes, it does. It looks like a new Formula, but I used my old commit, which wrote its own script. Fixed it.

File: Formula/procyon-decompiler.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      class T
+      {
+          public static void main(final String[] array) {
+              System.out.println("Hello World!");
+          }
+      }
+    EOS
+    (testpath/"T.java").write fixture
+    system "#{Formula["openjdk"].bin}/javac", "T.java"
+    fixture.match pipe_output("#{bin}/procyon-decompiler", "T.class")
```
Comment (carlocab) [inline Formula/procyon-decompiler.rb:26]:
```suggestion
    assert_match fixture, pipe_output(bin/"procyon-decompiler", "T.class")
```

Comment (middagj) [reply Formula/procyon-decompiler.rb]: That change revealed a bug in the test, first argument should be the cmd array. Fixed.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**